control/init.sh

```diff
@@ -50,7 +50,13 @@ if [ -z ${TRAVIS_TEST} ]; then
     source virtualenvwrapper.sh
     if [ ! -d ~/.virtualenvs/ansible ]; then
         echo "Creating ansible virtualenv..."
-        mkvirtualenv ansible
+        mkvirtualenv ansible --python $(which python2)
+    # If we by mistake are using a py3 env (unsupported), then replace with a py2 env
+    elif ~/.virtualenvs/ansible/bin/python -c 'print ""' 2> /dev/null; [ "$?" -ne "0" ]; then
+        echo "Replacing ansible py3 virtualenv with py2 virtualenv..."
+        deactivate 2> /dev/null || :  # deactivate if in a virtualenv, else ignore error
+        rmvirtualenv ansible
```

Comment:
This is a destructive operation, and this script is usually run automatically on login. This seems like it could cause great inconvenience and confusion if someone has purposefully configured their virtualenv with Python 3. What do you think of displaying a loud warning that things are unlikely to work rather than deleting their virtualenv and automatically setting up a new one?

Reply:
Currently a py3 virtualenv doesn't work at all for commcare-cloud, which is why I felt comfortable deleting it. Presumably we will work on this in the future, and at _that_ point, I could see this tripping up whoever's working on setting us up for the switch.

Seeing as I just added this for specific problem we were having where I'd inadvertently set up the new staging control with py3 virtualenvs and needed to reset that, this code's specific reason for existing is kind of over. So yeah I'm happy to replace it with a warning and call it a day.

Reply:
https://github.com/dimagi/commcare-cloud/pull/2832